packages/core/src/domain/configuration/configuration.ts

```diff
@@ -95,12 +97,44 @@ export type Configuration = typeof DEFAULT_CONFIGURATION &
   TransportConfiguration & {
     cookieOptions: CookieOptions
 
-    service?: string
-    beforeSend?: BeforeSendCallback
+    service: string | undefined
+    beforeSend: BeforeSendCallback | undefined
 
     actionNameAttribute?: string
   }
 
+export function validateAndBuildConfiguration(
+  initConfiguration: InitConfiguration,
+  buildEnv: BuildEnv
+): Configuration | undefined {
+  if (!initConfiguration || !initConfiguration.clientToken) {
+    display.error('Client Token is not configured, we will not send any data.')
+    return
+  }
+
+  // Set the experimental feature flags as early as possible so we can use them in most places
+  updateExperimentalFeatures(initConfiguration.enableExperimentalFeatures)
+
+  const configuration: Configuration = {
+    beforeSend:
+      initConfiguration.beforeSend && catchUserErrors(initConfiguration.beforeSend, 'beforeSend threw an error:'),
+    cookieOptions: buildCookieOptions(initConfiguration),
+    service: initConfiguration.service,
+    ...computeTransportConfiguration(initConfiguration, buildEnv),
+    ...DEFAULT_CONFIGURATION,
```

Comment:
Should we trim the default configuration used here as well?
or is it planned for later?

Reply:
It is planned for later, when I move RUM-only options to the RUM package.